app/models/git_repository.rb

```diff
@@ -16,10 +16,15 @@ class GitRepository < ApplicationRecord
 
   has_many :git_branches, :dependent => :destroy
   has_many :git_tags, :dependent => :destroy
-  after_destroy :delete_repo_dir # TODO: Need to distribute this to all systems
+  after_destroy :broadcast_repo_dir_delete
 
   INFO_KEYS = %w(commit_sha commit_message commit_time name).freeze
 
+  def self.delete_repo_dir(id, directory_name)
```

Comment:
Minor, but you could leave this as an instance method and use `:instance_id => id` in the `MiqQueue.broadcast` call.

I think a class method that takes the object id to log it is a bit weird. Plus `delete_repo_dir` is a nice method to have on the instance generally.

Reply:
Oh, nevermind. After delete, duh. Ignore this.